packages/flutter/test/material/chip_test.dart

```diff
@@ -1585,10 +2229,93 @@ void main() {
     expect(getDeleteDrawerProgress(tester), equals(0.0));
   });
 
-  testWidgets('Selection without avatar works as expected on RawChip', (WidgetTester tester) async {
+  testWidgets('Material3 - Selection with avatar works as expected on RawChip', (WidgetTester tester) async {
     bool selected = false;
     final UniqueKey labelKey = UniqueKey();
-    Future<void> pushChip({ bool selectable = false }) async {
+    Future<void> pushChip({ Widget? avatar, bool selectable = false }) async {
+      return tester.pumpWidget(
+        wrapForChip(
+          child: Wrap(
+            children: <Widget>[
+              StatefulBuilder(builder: (BuildContext context, StateSetter setState) {
+                return RawChip(
+                  avatar: avatar,
+                  onSelected: selectable
+                    ? (bool value) {
+                        setState(() {
+                          selected = value;
+                        });
+                      }
+                    : null,
+                  selected: selected,
+                  label: Text('Long Chip Label', key: labelKey),
+                  shape: const StadiumBorder(),
+                );
+              }),
+            ],
+          ),
+        ),
+      );
+    }
+
+    // With avatar, but not selectable.
+    final UniqueKey avatarKey = UniqueKey();
+    await pushChip(
+      avatar: SizedBox(width: 40.0, height: 40.0, key: avatarKey),
+    );
+    expect(tester.getSize(find.byType(RawChip)), equals(const Size(265.5, 48.0)));
+
+    // Turn on selection.
+    await pushChip(
+      avatar: SizedBox(width: 40.0, height: 40.0, key: avatarKey),
+      selectable: true,
+    );
+    await tester.pumpAndSettle();
+
+    expect(SchedulerBinding.instance.transientCallbackCount, equals(0));
+
+    // Simulate a tap on the label to select the chip.
+    await tester.tap(find.byKey(labelKey));
+    expect(selected, equals(true));
+    expect(SchedulerBinding.instance.transientCallbackCount, equals(isCanvasKit ? 3 : 1));
```

Comment:
There are some test failures like this:
![Screenshot 2024-01-04 at 12 50 24 PM](https://github.com/flutter/flutter/assets/36861262/4af7663e-da3a-4425-82c7-bdcd4a415729)

I think we need to add `kIsWeb` check before `isCanvasKit`.
```suggestion
    expect(SchedulerBinding.instance.transientCallbackCount, equals(kIsWeb && isCanvasKit ? 3 : 1));
```

Reply:
You're right, i missed that! Thanks